api/utils/keys/piv/service.go

```diff
@@ -227,10 +242,10 @@ func (s *YubiKeyService) GetFullKeyRef(serialNumber uint32, slotKey hardwarekey.
 // Get the given YubiKey with the serial number. If the provided serialNumber is "0",
 // return the first YubiKey found in the smart card list.
 func (s *YubiKeyService) getYubiKey(serialNumber uint32) (*YubiKey, error) {
-	yubiKeysMux.Lock()
-	defer yubiKeysMux.Unlock()
+	s.yubiKeysMux.Lock()
+	defer s.yubiKeysMux.Unlock()
```

Comment:
Is this mux protecting `s.yubiKeys`, or is it protecting some global state used by `FindYubiKey`? Can you run `FindYubiKey` in parallel in two different `*YubiKeyService`s? Can you run `FindYubiKey` in parallel for the same serial number?

Reply:
Ah I see how it could look like that, but no the mutex is just protecting the yubikey map. `FindYubiKey` is independent.

Reply:
FYI this is exercised in `TestGetYubiKeyPrivateKey_Interactive`, though CI doesn't run it.